### PR TITLE
Allow developers to specify custom resolve options.

### PR DESCRIPTION
**Pull request by MLoughry**
Integrating my last change in #412, I ran into an issue because `@linaria/babel-preset` (at least at its current version of `3.0.0-beta.23`) throws an error when given an ESM module. Since we don't want to use CJS as the basis of our whole build, we need to specify custom `conditionNames` for this loader in particular.

### CODE REVIEW

**Comment by MLoughry** [comment]
@layershifter When you get a chance, can you please review?

File: packages/webpack-loader/src/webpackLoader.test.ts [hunk 1]
```diff
@@ -269,6 +269,18 @@ describe('webpackLoader', () => {
...
+  // Asserts that "webpackResolveOptions" are handled properly
+  testFixture('webpack-resolve-options', {
+    loaderOptions: {
+      webpackResolveOptions: { conditionNames: ['source', 'require'] },
+    },
+    webpackConfig: {
+      resolve: {
+        conditionNames: ['source', 'import'],
+      },
+    },
```
Comment (layershifter) [inline packages/webpack-loader/src/webpackLoader.test.ts:281]:
```suggestion
    loaderOptions: {
      webpackResolveOptions: {
        extensions: ['.ts', '.jsx'],
      },
    },
```

Let's make the assertion real as we don't have `package.json` files in this fixture to test against.

File: change/@griffel-webpack-loader-741e9ad7-6c83-458e-9add-a8541efa0939.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Add the ability to specify custom webpack resolve options",
```
Comment (layershifter) [inline change/@griffel-webpack-loader-741e9ad7-6c83-458e-9add-a8541efa0939.json:3]:
```suggestion
  "comment": "feat: add webpackResolveOptions to specify custom resolve options",
```

**Review by layershifter — APPROVED** [review]

**Comment by layershifter** [comment]
Released in `@griffel/webpack-loader@2.2.0` 🚢

**Pull request merged**